patches/openwrt/0007-kernel-bridge-Implement-MLD-Querier-wake-up-calls-Android-bug-workaround.patch

```diff
@@ -0,0 +1,914 @@
+++	[DEV_ATTR_MULTICAST_WAKEUPCALL] = { .name = "multicast_wakeupcall", .type = BLOBMSG_TYPE_INT32 },
++ 	[DEV_ATTR_MULTICAST_ROUTER] = { .name = "multicast_router", .type = BLOBMSG_TYPE_INT32 },
++ 	[DEV_ATTR_MULTICAST_FAST_LEAVE] = { .name = "multicast_fast_leave", . type = BLOBMSG_TYPE_BOOL },
++ 	[DEV_ATTR_MULTICAST] = { .name ="multicast", .type = BLOBMSG_TYPE_BOOL },
++@@ -261,6 +262,7 @@ device_merge_settings(struct device *dev
++ 	n->multicast = s->flags & DEV_OPT_MULTICAST ?
++ 		s->multicast : os->multicast;
++ 	n->multicast_to_unicast = s->multicast_to_unicast;
+++	n->multicast_wakeupcall = s->multicast_wakeupcall;
++ 	n->multicast_router = s->multicast_router;
++ 	n->multicast_fast_leave = s->multicast_fast_leave;
++ 	n->learning = s->learning;
++@@ -386,6 +388,11 @@ device_init_settings(struct device *dev,
++ 		s->flags |= DEV_OPT_MULTICAST_TO_UNICAST;
++ 	}
++ 
+++	if ((cur = tb[DEV_ATTR_MULTICAST_WAKEUPCALL])) {
+++		s->multicast_wakeupcall = blobmsg_get_u32(cur);
+++		s->flags |= DEV_OPT_MULTICAST_WAKEUPCALL;
+++	}
+++
++ 	if ((cur = tb[DEV_ATTR_MULTICAST_ROUTER])) {
++ 		s->multicast_router = blobmsg_get_u32(cur);
++ 		if (s->multicast_router <= 2)
++@@ -1186,6 +1193,8 @@ device_dump_status(struct blob_buf *b, s
++ 			blobmsg_add_u32(b, "dadtransmits", st.dadtransmits);
++ 		if (st.flags & DEV_OPT_MULTICAST_TO_UNICAST)
++ 			blobmsg_add_u8(b, "multicast_to_unicast", st.multicast_to_unicast);
+++		if (st.flags & DEV_OPT_MULTICAST_WAKEUPCALL)
+++			blobmsg_add_u32(b, "multicast_wakeupcall", st.multicast_wakeupcall);
++ 		if (st.flags & DEV_OPT_MULTICAST_ROUTER)
++ 			blobmsg_add_u32(b, "multicast_router", st.multicast_router);
++ 		if (st.flags & DEV_OPT_MULTICAST_FAST_LEAVE)
++--- a/device.h
+++++ b/device.h
++@@ -44,6 +44,7 @@ enum {
++ 	DEV_ATTR_NEIGHREACHABLETIME,
++ 	DEV_ATTR_DADTRANSMITS,
++ 	DEV_ATTR_MULTICAST_TO_UNICAST,
+++	DEV_ATTR_MULTICAST_WAKEUPCALL,
++ 	DEV_ATTR_MULTICAST_ROUTER,
++ 	DEV_ATTR_MULTICAST_FAST_LEAVE,
++ 	DEV_ATTR_MULTICAST,
++@@ -109,23 +110,24 @@ enum {
++ 	DEV_OPT_MTU6			= (1ULL << 12),
++ 	DEV_OPT_DADTRANSMITS		= (1ULL << 13),
++ 	DEV_OPT_MULTICAST_TO_UNICAST	= (1ULL << 14),
++-	DEV_OPT_MULTICAST_ROUTER	= (1ULL << 15),
++-	DEV_OPT_MULTICAST		= (1ULL << 16),
++-	DEV_OPT_LEARNING		= (1ULL << 17),
++-	DEV_OPT_UNICAST_FLOOD		= (1ULL << 18),
++-	DEV_OPT_NEIGHGCSTALETIME	= (1ULL << 19),
++-	DEV_OPT_MULTICAST_FAST_LEAVE	= (1ULL << 20),
++-	DEV_OPT_SENDREDIRECTS		= (1ULL << 21),
++-	DEV_OPT_NEIGHLOCKTIME		= (1ULL << 22),
++-	DEV_OPT_ISOLATE			= (1ULL << 23),
++-	DEV_OPT_IP6SEGMENTROUTING	= (1ULL << 24),
++-	DEV_OPT_DROP_V4_UNICAST_IN_L2_MULTICAST = (1ULL << 25),
++-	DEV_OPT_DROP_V6_UNICAST_IN_L2_MULTICAST = (1ULL << 26),
++-	DEV_OPT_DROP_GRATUITOUS_ARP	= (1ULL << 27),
++-	DEV_OPT_DROP_UNSOLICITED_NA	= (1ULL << 28),
++-	DEV_OPT_ARP_ACCEPT		= (1ULL << 29),
++-	DEV_OPT_SPEED			= (1ULL << 30),
++-	DEV_OPT_DUPLEX			= (1ULL << 31),
+++	DEV_OPT_MULTICAST_WAKEUPCALL	= (1ULL << 15),
```

Comment:
Potential future cleanup (for the next big rebase?):

Moving all these enum fields by 1 seems unnecessary. Maybe add the new flag at `1ULL << 63` instead?